src/toolbox/shamir.rs

```diff
@@ -303,8 +314,9 @@ mod tests {
 
     #[test]
     fn shamir_reconstruct_bad_threshold() {
-        let shares = vec![Scalar::from(272u32), Scalar::from(935u32), Scalar::from(1990u32)];
-        match SecretShare::reconstruct(4, shares) {
+        let shares = vec![Some(Scalar::from(272u32)), Some(Scalar::from(935u32)), Some(Scalar::from(1990u32))];
+        let mut sham = Shamir::new_without_rng(4);
+        match sham.reconstruct(&shares) {
```

Comment:
Minor thing but I guess you could do this in one line:
`assert!(e.is_err() && e.contains("Not enough shares"), "Should not have been able to reconstruct!");`

Reply:
Those don't mean the same thing, though.  In the shorter version, if some other error is thrown, the message given will still indicate that reconstruction was successful.  The two error conditions are distinct enough (as I found during development) that I wanted to keep the messages separate.